src/modules/rating/rating.ts

```diff
@@ -0,0 +1,173 @@
+///<reference path="../../../typings/angularjs/angular.d.ts"/>
+
+'use strict';
+
+const ratingConfig = {
+  max: 5,
+  size: '',
+  stateActive: 'active',
+  stateHover: 'selected',
+  stateHoverParent: 'selected',
+  type: 'star'
+};
+
+class SmRatingController {
+  static $inject = ['$scope', '$element', '$attrs'];
+
+  hoverValue: number;
+  readonly: boolean;
+
+  stateHoverParent: string;
+
+  ratingStates: RatingState[];
+  icons: Icon[];
+
+  ngModel: ng.INgModelController;
+
+  constructor(public $scope: ng.IScope, public $element: ng.IAugmentedJQuery, public $attrs) {
+    const
+      ratingStates: RatingState[] = this.evalAttribute('ratingStates') || Array.apply(null, Array(this.evalAttribute('max'))),
+      stateActive = this.evalAttribute('stateActive'),
+      stateHover = this.evalAttribute('stateHover'),
+      stateHoverParent = this.evalAttribute('stateHoverParent');
+
+    if (angular.isDefined($attrs.readonly)) {
+      $scope.$watch($attrs.readonly, (readonly: boolean) => {
+        this.readonly = readonly;
+      });
+    }
+
+    this.icons = ratingStates.map((state, index): Icon => {
+      const
+        iconElm = angular.element('<i class="icon"></i>'),
+        value = index + 1;
+
+      iconElm.on('mouseenter', () => {
+        $scope.$apply(() => {
+          if (!this.readonly) {
+            this.hoverValue = value;
+            this.$element.addClass(stateHoverParent);
+            this.updateStateHover();
+          }
+          if (angular.isDefined(this.$attrs.onHover)) {
+            this.$scope.$eval(this.$attrs.onHover, { value: value });
+          }
+        });
+      });
+
+      iconElm.on('click', () => {
+        this.rate(value);
+      });
+
+      return angular.extend(<Icon>{
+        element: iconElm,
+        index: index,
+        stateActive: stateActive,
+        stateHover: stateHover
+      }, state);
+    });
+
+    this.icons.forEach(icon => {
+      $element.append(icon.element);
+    });
+
+    $element
+      .addClass('ui rating')
+      .addClass($attrs.type || ratingConfig.type)
+      .addClass($attrs.size || ratingConfig.size)
+      .attr({
+        'aria-valuemax': this.icons.length,
+        'aria-valuemin': 0,
+        role: 'slider',
+        tabindex: 0
+      })
+      .on('mouseleave', () => {
+        this.hoverValue = -1;
+        if (angular.isDefined($attrs.onLeave)) {
+          $scope.$evalAsync($attrs.onLeave);
+        }
+        $element.removeClass(stateHoverParent);
+        this.updateStateHover();
+      })
+      .on('keydown', (evt: any) => {
+        if (/(37|38|39|40)/.test(evt.which)) {

```

Comment:
This part should be more generic, probably something like [this](https://github.com/angular/angular/blob/master/modules/angular2_material/src/core/key_codes.ts). That's fine for now